src/main/java/games/strategy/debug/GenericConsole.java

```diff
@@ -63,7 +64,7 @@ public void dumpStacks() {
    */
   protected void displayStandardError() {
     final SynchedByteArrayOutputStream out = new SynchedByteArrayOutputStream(System.err);
-    final ThreadReader reader = new ThreadReader(out, textArea, true, getConsoleInstance());
+    final ThreadReader reader = new ThreadReader(out, textArea, Boolean.TRUE::booleanValue, getConsoleInstance());
```

Comment:
So you're basically taking the Boolean object field,converting it to the primitive type which is then auto casted back to an object because generics don't support primitives.
Perhaps there's a better way like `Boolean::TRUE` or something (I don't think this syntax is even valid)

Reply:
`Boolean.TRUE::booleanValue` is just a method reference to the `Boolean#booleanValue()` method bound to the `Boolean.TRUE` instance.  I don't believe there is any boxing/unboxing or casting involved in obtaining this method reference.  It's basically equivalent to `() -> true`.

However, assuming no JIT optimizations, there is boxing/unboxing when invoking the method reference.  The primitive `boolean` returned from `Boolean#booleanValue()` will need to be boxed to a `Boolean` and then subsequently unboxed back to a primitive `boolean` in the conditional check within `ThreadReader`.

But, I think your point is I should have used `BooleanSupplier` instead of `Supplier<Boolean>` to avoid the boxing.  I'll change that. :smile: